### PR TITLE
ddl: Add the txn entry limit

### PR DESCRIPTION
**Pull request by zimulala**
It will exceed the transaction limit when the table schema info is too large. 
Here if this job is large we need to cancel this job.

### CODE REVIEW

**Comment by zimulala** [comment]
PTAL @winkyao @coocood @breeswish

File: ddl/ddl_worker.go [hunk 1]
```diff
@@ -97,6 +97,23 @@ func (d *ddl) getFirstDDLJob(t *meta.Meta) (*model.Job, error) {
...
 
+// handleTooLargeJob handles the too large DDL job.
+func (d *ddl) handleTooLargeJob(t *meta.Meta, job *model.Job, err error) error {
+	if err == nil {
+		return nil
+	}
+	if kv.ErrEntryTooLarge.Equal(err) {
+		log.Warnf("[ddl] update DDL job %v failed %v", job, errors.ErrorStack(err))
+		// Reduce this txn entry size.
+		job.BinlogInfo.Clean()
```
Comment (winkyao) [inline ddl/ddl_worker.go]: why need to clean binloginfo here?

Comment (zimulala) [reply ddl/ddl_worker.go]: Because binloginfo's tableInfo is large. If doesn't clean it, we can't commit this job. @winkyao

Comment (winkyao) [reply ddl/ddl_worker.go]: got it.

File: ddl/ddl_worker.go [hunk 2]
```diff
@@ -97,6 +97,23 @@ func (d *ddl) getFirstDDLJob(t *meta.Meta) (*model.Job, error) {
 	return job, errors.Trace(err)
 }
 
+// handleTooLargeJob handles the too large DDL job.
+func (d *ddl) handleTooLargeJob(t *meta.Meta, job *model.Job, err error) error {
```
Comment (coocood) [inline ddl/ddl_worker.go]: May be `handleUpdateJobError` is better, probably other errors need to be handled in the future.

**Comment by zimulala** [comment]
PTAL @coocood

**Comment by coocood** [comment]
LGTM

**Pull request merged**